### DISCUSSION TITLE
Playing audio CD on Linux

### DISCUSSION THREAD

**Discussion by Jercik** [avantrec/soco-cli#51] [Q&A]
Hi!

I would like to create DIY Sonos audio CD player. My idea is to play audio CD from a USB CD drive connected to a Raspberry Pi and stream the audio to my Sonos speakers.

I tried playing audio CD from my Mac using `soco-cli` and it worked fine, but the documentation mentions it will not work on Linux. My immediate thought is to split the problem into two parts: play the CD using eg. `cdtool` or `cdcd` and somehow route the audio to `soco-cli` and stream it to Sonos.

Does it make sense? Is this something that can be archived with `soco-cli`?

**Comment by pwt** [comment]
Hi @Jercik 

I don't think it would be easy to do it the way you're suggesting with SoCo-CLI. It's not designed to re-route audio.

However, if you could mount the CD in a way that its files are exposed (as .WAV files?) and readable then SoCo-CLI's `play_file` functions should be able to play these files on Sonos speakers.

I wonder whether `gio` would support this. E.g., see: https://superuser.com/questions/1627938/mounting-an-audio-cd-in-a-linux-terminal-or-in-a-bash-script.

**Reply by Jercik** [reply]
Thank you! I'll be sure to test it out.

Meanwhile, I stumbled upon this gist: https://gist.github.com/gwpl/4efd5933fbf9015ee36cd0e443bfea39. It uses pulseaudio and cvlc to stream system audio over html and it employs the `add_uri_to_queue()` function from https://github.com/SoCo/SoCo to initiate the stream on Sonos speakers. I'm curious if there's a way to achieve a similar outcome using SoCo-CLI to direct Sonos to play a stream? Is SoCo-CLI using https://github.com/SoCo/SoCo under the hood? Thanks in advance!

**Comment by pwt** [comment]
Yes, SoCo-CLI is a command line wrapper for SoCo.

The `play_uri` action is natively supported by SoCo-CLI: see the documentation in https://github.com/avantrec/soco-cli#playback-control.
